### ISSUE TITLE
FreeRDP crash with OpenSSL v3 (Bugzilla Bug 2672)

### ISSUE THREAD

**Issue by tpgxyz** [OpenMandrivaAssociation/test2#2672]
This issue was created automatically with bugzilla2github
# Bugzilla Bug 2672
Date: 2021-01-23 16:54:27 +0000
From: @mandian
To:   @berolinux
CC:   angrypenguinpoland@gmail.com, @benbullard79, bugs@openmandriva.org, @mandian, nix.or.die@gmail.com, nobodydead@gmail.com, @tpgxyz

Depends on:   #2670
Last updated: 2021-09-14 01:44:18 +0000

## Comment 20556
Date: 2021-01-23 16:54:27 +0000
From: @mandian

Hi,

Remmina crash every time I try to connect with RDP protocol. I guess Remmina in not fully compatible with openssl v3.

Here is the core dump:

           PID: 113319 (remmina)
           UID: 1000 (mandian)
           GID: 1000 (mandian)
        Signal: 11 (SEGV)
     Timestamp: Sat 2021-01-23 00:00:00 GMT
  Command Line: remmina
    Executable: /usr/bin/remmina
 Control Group: /user.slice/user-1001.slice/session-2.scope
          Unit: session-2.scope
         Slice: user-1001.slice
       Session: 2
     Owner UID: 1000 (mandian)
       Boot ID: acf7a98980080ca9089098f0e098d98a
    Machine ID: e798afb907af34af7fa0a8be9904c098
      Hostname: mandian-pc
       Storage: /var/lib/systemd/coredump/core.remmina.1001.c972db596f8f424b86c82ba5b136a32e.113319.1611418735000000.zst
       Message: Process 113319 (remmina) of user 1000 dumped core.
                
                Stack trace of thread 116849:
                #0  0x00007f0c338d8b0e EVP_CIPHER_CTX_set_key_length (libcrypto.so.3 + 0x2dfb0e)
                #1  0x00007f0c2bdb0c86 winpr_RC4_New_Internal (libwinpr2.so.2 + 0xa3c86)
                #2  0x00007f0c2bde33ef ntlm_rc4k (libwinpr2.so.2 + 0xd63ef)
                #3  0x00007f0c2bde4241 ntlm_read_ChallengeMessage (libwinpr2.so.2 + 0xd7241)
                #4  0x00007f0c2bde6fdc ntlm_InitializeSecurityContextW (libwinpr2.so.2 + 0xd9fdc)
                #5  0x00007f0c2bde59c3 ntlm_InitializeSecurityContextA (libwinpr2.so.2 + 0xd89c3)
                #6  0x00007f0c2bde79a6 negotiate_InitializeSecurityContextA (libwinpr2.so.2 + 0xda9a6)
                #7  0x00007f0c2bdec64f winpr_InitializeSecurityContextA (libwinpr2.so.2 + 0xdf64f)
                #8  0x00007f0c2bf4c604 nla_recv_pdu (libfreerdp2.so.2 + 0xa2604)
                #9  0x00007f0c2bf7e137 rdp_recv_callback (libfreerdp2.so.2 + 0xd4137)
                #10 0x00007f0c2bf83934 rdp_check_fds (libfreerdp2.so.2 + 0xd9934)
                #11 0x00007f0c2bf71a30 rdp_client_connect (libfreerdp2.so.2 + 0xc7a30)
                #12 0x00007f0c2bf65333 freerdp_connect (libfreerdp2.so.2 + 0xbb333)
                #13 0x00007f0c3006e6b3 remmina_rdp_main_thread (remmina-plugin-rdp.so + 0x106b3)
                #14 0x00007f0c33c0ffc8 start_thread (libpthread.so.0 + 0x8fc8)
                #15 0x00007f0c33388f4f __clone (libc.so.6 + 0x112f4f)
                
                Stack trace of thread 113342:
                #0  0x00007f0c3337d51f __poll (libc.so.6 + 0x10751f)
                #1  0x00007f0c34109d65 g_poll (libglib-2.0.so.0 + 0xf5d65)
                #2  0x00007f0c340f817c g_main_context_iterate (libglib-2.0.so.0 + 0xe417c)
                #3  0x00007f0c340fa4fa glib_worker_main (libglib-2.0.so.0 + 0xe64fa)
                #4  0x00007f0c3412a0df g_thread_proxy (libglib-2.0.so.0 + 0x1160df)
                #5  0x00007f0c3415acc4 linux_pthread_proxy (libglib-2.0.so.0 + 0x146cc4)
                #6  0x00007f0c33c0ffc8 start_thread (libpthread.so.0 + 0x8fc8)
                #7  0x00007f0c33388f4f __clone (libc.so.6 + 0x112f4f)
                
                Stack trace of thread 113343:
                #0  0x00007f0c3337d51f __poll (libc.so.6 + 0x10751f)
                #1  0x00007f0c34109d65 g_poll (libglib-2.0.so.0 + 0xf5d65)
                #2  0x00007f0c340f817c g_main_context_iterate (libglib-2.0.so.0 + 0xe417c)
                #3  0x00007f0c340f85cf g_main_loop_run (libglib-2.0.so.0 + 0xe45cf)
                #4  0x00007f0c33fd2998 gdbus_shared_thread_func (libgio-2.0.so.0 + 0x1d2998)
                #5  0x00007f0c3412a0df g_thread_proxy (libglib-2.0.so.0 + 0x1160df)
                #6  0x00007f0c3415acc4 linux_pthread_proxy (libglib-2.0.so.0 + 0x146cc4)
                #7  0x00007f0c33c0ffc8 start_thread (libpthread.so.0 + 0x8fc8)
                #8  0x00007f0c33388f4f __clone (libc.so.6 + 0x112f4f)
                
                Stack trace of thread 116850:
                #0  0x00007f0c3337d51f __poll (libc.so.6 + 0x10751f)
                #1  0x00007f0c2bda236d WaitForSingleObject (libwinpr2.so.2 + 0x9536d)
                #2  0x00007f0c2be75638 play_thread (libfreerdp-client2.so.2 + 0x3c638)
                #3  0x00007f0c2bddf4e5 thread_launcher (libwinpr2.so.2 + 0xd24e5)
                #4  0x00007f0c33c0ffc8 start_thread (libpthread.so.0 + 0x8fc8)
                #5  0x00007f0c33388f4f __clone (libc.so.6 + 0x112f4f)
                
                Stack trace of thread 116851:
                #0  0x00007f0c3337d51f __poll (libc.so.6 + 0x10751f)
                #1  0x00007f0c2bda2843 WaitForMultipleObjects (libwinpr2.so.2 + 0x95843)
                #2  0x00007f0c2bdddcc1 thread_pool_work_func (libwinpr2.so.2 + 0xd0cc1)
                #3  0x00007f0c2bddf4e5 thread_launcher (libwinpr2.so.2 + 0xd24e5)
                #4  0x00007f0c33c0ffc8 start_thread (libpthread.so.0 + 0x8fc8)
                #5  0x00007f0c33388f4f __clone (libc.so.6 + 0x112f4f)
                
                Stack trace of thread 116853:
                #0  0x00007f0c3337d51f __poll (libc.so.6 + 0x10751f)
                #1  0x00007f0c2bda2843 WaitForMultipleObjects (libwinpr2.so.2 + 0x95843)
                #2  0x00007f0c2bdddcc1 thread_pool_work_func (libwinpr2.so.2 + 0xd0cc1)
                #3  0x00007f0c2bddf4e5 thread_launcher (libwinpr2.so.2 + 0xd24e5)
                #4  0x00007f0c33c0ffc8 start_thread (libpthread.so.0 + 0x8fc8)
                #5  0x00007f0c33388f4f __clone (libc.so.6 + 0x112f4f)
                
                Stack trace of thread 116852:
                #0  0x00007f0c3337d51f __poll (libc.so.6 + 0x10751f)
                #1  0x00007f0c2bda2843 WaitForMultipleObjects (libwinpr2.so.2 + 0x95843)
                #2  0x00007f0c2bdddcc1 thread_pool_work_func (libwinpr2.so.2 + 0xd0cc1)
                #3  0x00007f0c2bddf4e5 thread_launcher (libwinpr2.so.2 + 0xd24e5)
                #4  0x00007f0c33c0ffc8 start_thread (libpthread.so.0 + 0x8fc8)
                #5  0x00007f0c33388f4f __clone (libc.so.6 + 0x112f4f)
                
                Stack trace of thread 116854:
                #0  0x00007f0c3337d51f __poll (libc.so.6 + 0x10751f)
                #1  0x00007f0c2bda2843 WaitForMultipleObjects (libwinpr2.so.2 + 0x95843)
                #2  0x00007f0c2bdddcc1 thread_pool_work_func (libwinpr2.so.2 + 0xd0cc1)
                #3  0x00007f0c2bddf4e5 thread_launcher (libwinpr2.so.2 + 0xd24e5)
                #4  0x00007f0c33c0ffc8 start_thread (libpthread.so.0 + 0x8fc8)
                #5  0x00007f0c33388f4f __clone (libc.so.6 + 0x112f4f)
                
                Stack trace of thread 113319:
                #0  0x00007f0c3337d51f __poll (libc.so.6 + 0x10751f)
                #1  0x00007f0c34109d65 g_poll (libglib-2.0.so.0 + 0xf5d65)
                #2  0x00007f0c340f817c g_main_context_iterate (libglib-2.0.so.0 + 0xe417c)
                #3  0x00007f0c340f829d g_main_context_iteration (libglib-2.0.so.0 + 0xe429d)
                #4  0x00007f0c33f83ebd g_application_run (libgio-2.0.so.0 + 0x183ebd)
                #5  0x0000000000233c39 main (remmina + 0x33c39)
                #6  0x00007f0c3329ed3a __libc_start_main (libc.so.6 + 0x28d3a)
                #7  0x000000000023224a _start (remmina + 0x3224a)

## Comment 20559
Date: 2021-01-23 20:06:40 +0000
From: @benbullard79

*** Bug #2673 has been marked as a duplicate of this bug. ***

## Comment 20560
Date: 2021-01-23 22:17:34 +0000
From: @mandian

Well, I investigate a little further and I suspect the culprit is the library libwinpr part of freeRDP so it may affect all the other softer using it fr RDP too.

## Comment 20577
Date: 2021-02-03 11:41:38 +0000
From: @tpgxyz

I updated to new version 1.4.11 please try it.

## Comment 20597
Date: 2021-02-12 23:03:44 +0000
From: @mandian

Hi TPG,

the issues is still here. Remmina works fine but FreeRDP is not fully compatible with OpenSSL v3. I modified the post title according.

## Comment 20605
Date: 2021-03-02 12:00:30 +0000
From: @tpgxyz

Looks like we are hit by OpenSSLv3 alpha very hard :<

## Comment 20606
Date: 2021-03-02 14:06:32 +0000
From: @tpgxyz

We have a potential fix in opensslv3
https://github.com/openssl/openssl/issues/14392

## Comment 20607
Date: 2021-03-02 14:13:23 +0000
From: @tpgxyz

There is also a suggestion from opnessl guys to upgrade FreeRDP at first.

## Comment 20629
Date: 2021-03-08 13:08:04 +0000
From: AngryPenguin &lt;<angrypenguinpoland@gmail.com>&gt;

I updated FreeRDP and Remmina to new version.
Please recheck now.

## Comment 20635
Date: 2021-04-18 19:08:55 +0000
From: @mandian

Hi,
nothing changed here, I got the same crash and coredump.

## Comment 20713
Date: 2021-06-01 22:21:22 +0000
From: @tpgxyz

Fixed openssl-3.0.0-alpha16 has been released for 4.2.
Please update and give it a retry.

## Comment 20741
Date: 2021-06-14 20:50:45 +0000
From: @benbullard79

Need to find out if this is fixed or not?(In reply to Tomasz Paweł Gajc from comment #10)
> Fixed openssl-3.0.0-alpha16 has been released for 4.2.
> Please update and give it a retry.

Need to find out if this is fixed or not?

## Comment 20750
Date: 2021-06-26 23:15:00 +0000
From: @mandian

Hi Ben,

I am sorry for the delay but the remote host I use with RDP is down this month. I'll try ASAP on July.

## Comment 20757
Date: 2021-07-22 17:51:55 +0000
From: @mandian

Hi Ben,

I tried again but the problem is still here.

## Comment 20771
Date: 2021-09-14 01:44:18 +0000
From: @benbullard79

(In reply to mandian from comment #13)
> Hi Ben,
> 
> I tried again but the problem is still here.

OK

**Comment by mandian** [comment]
freerdp 2.4.1 + remmina 1.4.21 fixed it.

**Comment by benbullard79** [comment]
Guess this should be closed then.
